src/main.c

```diff
@@ -589,8 +624,13 @@ static struct window* window_create(const char* app_id, const char* title)
 	w->preferred_buffer_scale = 0;
 	pixman_region_init(&w->current_damage);
 
-	w->wl_bg_buffer = wp_single_pixel_buffer_manager_v1_create_u32_rgba_buffer(
-			single_pixel_manager, 0, 0, 0, UINT32_MAX);
+	if (single_pixel_manager)
+		w->wl_bg_buffer = wp_single_pixel_buffer_manager_v1_create_u32_rgba_buffer(
+				single_pixel_manager, 0, 0, 0, UINT32_MAX);
+	else
+		create_single_pixel_buffer_fallback(&w->wl_bg_pixels, &w->wl_bg_buffer,
```

Comment:
Why make `wl_bg_buffer` an argument here instead of assigning the returned value to it?

Reply:
Because we need to assign both bg_pixels and bg_buffer I assigned both with pointers. But we could also use return for one of them.